File: app/src/main/java/ru/ls/donkitchen/activity/receiptdetail/ReceiptDetail.java
package ru.ls.donkitchen.activity.receiptdetail;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;

import ru.ls.donkitchen.activity.base.BaseActivity;

/**
 * @author Lord (Kuleshov M.V.)
 * @since 13.10.15
 */
public class ReceiptDetail extends BaseActivity {
	public static final String EXT_IN_RECEIPT_ID = "receipt_id";
	public static final String EXT_IN_RECEIPT_NAME = "receipt_name";

	int receiptId;
	String receiptName;

	@Override
	protected Fragment loadFragment() {
		return ReceiptDetailFragment.newInstance(receiptId, receiptName);
	}

	@Override
	protected void readArguments(@NonNull Bundle args) {
		receiptId = args.getInt(EXT_IN_RECEIPT_ID, 0);
		receiptName = args.getString(EXT_IN_RECEIPT_NAME);
	}
}

File: app/src/main/kotlin/ru/ls/donkitchen/activity/receiptdetail/ReceiptDetail.kt
package ru.ls.donkitchen.activity.receiptdetail

import android.os.Bundle
import com.arellomobile.mvp.MvpAppCompatFragment
import ru.ls.donkitchen.activity.base.BaseActivity

/**
 *  Просмотр рецепта
 *
 * @author Lord (Kuleshov M.V.)
 * @since 17.10.16
 */
class ReceiptDetail: BaseActivity() {
    companion object {
        const val EXT_IN_RECEIPT_ID = "receipt_id"
        const val EXT_IN_RECEIPT_NAME = "receipt_name"
        const val EXT_IN_RECEIPT_PHOTO_URL = "receipt_photo_url"
    }

    var receiptId: Int = 0
    var receiptName: String = ""
    var receiptPhotoUrl: String = ""

    override fun loadFragment() = ReceiptDetailFragment.newInstance(receiptId, receiptName, receiptPhotoUrl)

    override fun readArguments(args: Bundle) {
        receiptId = args.getInt(EXT_IN_RECEIPT_ID, 0)
        receiptName = args.getString(EXT_IN_RECEIPT_NAME)
        receiptPhotoUrl = args.getString(EXT_IN_RECEIPT_PHOTO_URL)
    }
}
